src/SQL/Builder/DefaultWithSQLBuilder.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\SQL\Builder;
+
+use Doctrine\DBAL\Query\With;
+use Doctrine\DBAL\Query\WithQuery;
+
+use function array_map;
+use function count;
+use function implode;
+use function sprintf;
+
+final class DefaultWithSQLBuilder implements WithSQLBuilder
+{
+    public function buildSQL(WithQuery $query): string
+    {
+        $parts = array_map(
+            static fn (With $part) => sprintf(
+                '%s%s AS (%s)',
+                $part->name,
+                self::fields($part->fields),
+                $part->query,
+            ),
+            $query->withParts(),
+        );
+
+        return 'WITH ' . implode(', ', $parts) . ' ';
+    }
+
+    /** @param string[] $fields */
+    private static function fields(array $fields): string
```

Comment:
Why does this need to be a separate method and why static?

Reply:
Purpose was to make the array_map more readable. And static because it has not to be aware of the context of the class.
I can refactor this if this not acceptable.

Reply:
I don't think it adds readability. Furthermore, the code where the `sprintf` pattern contains a placeholder `%s` which may be populated with an empty string makes it hard to comprehend the logic.

The code would be much readable if all the code was contained in the same method, and instead of a ternary operator returning an empty string, there would be an `if` statement that would conditionally append a string. This would also contribute to more precise code coverage.